pkg/aws/cloudfront/options.go

```diff
@@ -9,15 +9,19 @@ import (
 
 type Option func(c *Client)
 
-func WithAwsRegion(region string) Option {
+func WithAWSRegion(region string) Option {
 	return func(c *Client) {
-		c.awsCfgOptions = append(c.awsCfgOptions, config.WithRegion(region))
+		if region != "" {
+			c.awsCfgOptions = append(c.awsCfgOptions, config.WithRegion(region))
+		}
 	}
 }
 
 func WithStaticCredentials(key string, secret string) Option {
 	return func(c *Client) {
-		c.awsCfgOptions = append(c.awsCfgOptions, config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider(key, secret, "")))
+		if key != "" && secret != "" {
+			c.awsCfgOptions = append(c.awsCfgOptions, config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider(key, secret, "")))
```

Comment:
I think there is a bug here that we missed in the initial PR.  If these options are called more than once, it will append another function to `awsCfgOptions`.    

I think `region` and `static credentials` can simply be private members of `Client` and the awsCfg options can just be constructed once.